cmd/tesla-http-proxy/main.go

```diff
@@ -100,14 +108,61 @@ func main() {
 	if err != nil {
 		return
 	}
-	p.Timeout = timeout
-	addr := fmt.Sprintf("%s:%d", host, port)
+	p.Timeout = httpConfig.timeout
+	addr := fmt.Sprintf("%s:%d", httpConfig.host, httpConfig.port)
 	log.Info("Listening on %s", addr)
 
 	// To add more application logic requests, such as alternative client authentication, create
 	// a http.HandleFunc implementation (https://pkg.go.dev/net/http#HandlerFunc). The ServeHTTP
 	// method of your implementation can perform your business logic and then, if the request is
 	// authorized, invoke p.ServeHTTP. Finally, replace p in the below ListenAndServeTLS call with
 	// an object of your newly created type.
-	log.Error("Server stopped: %s", http.ListenAndServeTLS(addr, certFilename, keyFilename, p))
+	log.Error("Server stopped: %s", http.ListenAndServeTLS(addr, httpConfig.certFilename, httpConfig.keyFilename, p))
+}
+
+// readConfig applies configuration from environment variables.
+// Values are not overwritten.
+func readFromEnvironment() error {
+	if httpConfig.certFilename == "" {
+		httpConfig.certFilename = os.Getenv("TESLA_TLS_CERT")
+	}
+
+	if httpConfig.keyFilename == "" {
+		httpConfig.keyFilename = os.Getenv("TESLA_TLS_KEY")
+	}
+
+	if httpConfig.host == "localhost" {
+		host, ok := os.LookupEnv("TESLA_HOST")
+		if ok {
+			httpConfig.host = host
+		}
+	}
+
+	if !httpConfig.verbose {
+		if verbose, ok := os.LookupEnv("TESLA_VERBOSE"); ok {
+			httpConfig.verbose = verbose != "false"
+		}
+	}
+
+	if httpConfig.port == defaultPort {
+		if port, ok := os.LookupEnv("TESLA_PORT"); ok {
+			var err error
+			httpConfig.port, err = strconv.Atoi(port)
+			if err != nil {
+				return fmt.Errorf("invalid port: %s", port)
+			}
+		}
+	}
+
+	if httpConfig.timeout == proxy.DefaultTimeout {
```

Comment:
The environment variable should either allow the same time formatting options as Golang's `time.Duration` parser, or the environment variable should be renamed `TESLA_TIMEOUT_SECONDS` or similar.